src/components/CourseInfo/CourseInfo.jsx

```diff
@@ -1,19 +1,35 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect } from 'react';
+import { useSelector } from 'react-redux';
 import { useParams, Link } from 'react-router-dom';
 import moment from 'moment';
-import { getCourseById } from '../../services/courseService';
+
 import dataFormat from '../../helpers/dataFormat';
-import styles from './styles.module.scss';
 import { ROUTE_COURSES } from '../../constants';
+import getCoursesSelector from '../../store/courses/selectors';
+import getAuthorsSelector from '../../store/authors/selectors';
+import { getAuthors, getCourseById } from '../../services';
+import getAuthorName from '../../helpers/getAuthorName';
+import styles from './styles.module.scss';
 
 const CourseInfo = () => {
-	const [course, setCourse] = useState(null);
+	const coursesData = useSelector(getCoursesSelector);
+	const course = coursesData.showCourse;
+
+	const authorsData = useSelector(getAuthorsSelector);
+	const availableAuthors = authorsData?.list;
+
 	const { courseId } = useParams();
 
 	useEffect(() => {
-		setCourse(getCourseById(courseId));
+		getCourseById(courseId);
```

Comment:
you can remove brackets for this effect and write it in one line
`useEffect(() => getCourseById(courseId), [courseId]);